mlflow/models/model.py

```diff
@@ -622,6 +625,7 @@ def to_dict(self) -> dict[str, Any]:
             res["databricks_runtime"] = databricks_runtime
         if self.signature is not None:
             res["signature"] = self.signature.to_dict()
+            res["_is_signature_from_type_hint"] = self.signature._is_signature_from_type_hint
```

Comment:
Q: do we have any other attributes in the dict representation starting with "_"? Otherwise, do we need to start with "_" specifically for this attribute? We don't need to expose Python convention to the serialized object.

Reply:
Yea we can remove the underscore :)